lib/omniauth/strategy.rb

```diff
@@ -208,9 +208,9 @@ def request_call # rubocop:disable CyclomaticComplexity, MethodLength, Perceived
         call_app!
       else
         if request.params['origin']
-          env['rack.session']['omniauth.origin'] = request.params['origin']
+          session['omniauth.origin'] = request.params['origin']
```

Comment:
This is not instance variable in origin code, I feel maybe there are some reason to not use instance variable (but author only knows...).